### DISCUSSION TITLE
How to elegantly implement this method ‘HashCallback’

### DISCUSSION THREAD

**Discussion by BinaryRyan** [FreeRDP/FreeRDP#10147] [Q&A] [answered]
I don't want to modify the source code of freerdp with strong invasiveness ways,But I couldn't find where to implement this callback.
thanks.

**Comment by akallabeth** [comment, marked as answer]
set it in `rdp_freerdp_peer` instance you create and it will be used. e.g. after a new connection was accepted with `PeerAccepted` you can set it there along with the argument you want the function to be called with.
the function itself then needs to generate/validate/... the credentials of the user connecting.

**Reply by BinaryRyan** [reply]
Okay, thanks.
